src/Umbraco.Web.BackOffice/Controllers/CodeFileController.cs

```diff
@@ -1,59 +1,66 @@
-﻿using System;
+﻿﻿using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using System.Net;
 using System.Net.Http;
-using System.Web.Http;
+using Microsoft.AspNetCore.Mvc;
 using Umbraco.Core;
-using Umbraco.Core.Cache;
-using Umbraco.Core.Configuration;
+ using Umbraco.Core.Configuration;
+ using Umbraco.Core.Configuration.Legacy;
 using Umbraco.Core.IO;
-using Umbraco.Core.Logging;
-using Umbraco.Core.Models;
-using Umbraco.Core.Persistence;
+ using Umbraco.Core.Mapping;
+ using Umbraco.Core.Models;
 using Umbraco.Core.Services;
-using Umbraco.Core.Strings;
-using Umbraco.Core.Strings.Css;
-using Umbraco.Web.Models.ContentEditing;
-using Umbraco.Web.Mvc;
-using Umbraco.Web.WebApi;
-using Umbraco.Web.WebApi.Filters;
-using Umbraco.Web.Trees;
+ using Umbraco.Core.Strings;
+ using Umbraco.Core.Strings.Css;
+ using Umbraco.Extensions;
+ using Umbraco.Web.Models.ContentEditing;
 using Stylesheet = Umbraco.Core.Models.Stylesheet;
 using StylesheetRule = Umbraco.Web.Models.ContentEditing.StylesheetRule;
-using Umbraco.Core.Mapping;
-using Umbraco.Web.Routing;
+using Umbraco.Web.BackOffice.Filters;
+ using Umbraco.Web.Common.ActionsResults;
+ using Umbraco.Web.Common.Attributes;
+using Umbraco.Web.Common.Exceptions;
+using Umbraco.Web.Editors;
 
-namespace Umbraco.Web.Editors
+namespace Umbraco.Web.BackOffice.Controllers
 {
     // TODO: Put some exception filters in our webapi to return 404 instead of 500 when we throw ArgumentNullException
     // ref: https://www.exceptionnotfound.net/the-asp-net-web-api-exception-handling-pipeline-a-guided-tour/
     [PluginController("UmbracoApi")]
-    [PrefixlessBodyModelValidator]
-    [UmbracoApplicationAuthorize(Core.Constants.Applications.Settings)]
+    //[PrefixlessBodyModelValidator]
+    [TypeFilter(typeof(UmbracoApplicationAuthorizeAttribute), Arguments = new object[]{new string []{Constants.Applications.Settings}})]
```

Comment:
I wonder if its possible to have another attribute inherit from TypeFilterAttribute so we can make this back to being `UmbracoApplicationAuthorizeAttribute` and have all the bits injected underneath since it's gonna be a pain to write this out everywhere